### PR TITLE
chore(storage/transfermanager): add usage metric headers

### PR DESCRIPTION
**Pull request by BrennaEpp**
- Also update storage to merge the header

### CODE REVIEW

File: storage/transfermanager/downloader.go [hunk 1]
```diff
@@ -306,3 +310,16 @@ type DownloadOutput struct {
...
+const (
+	xGoogHeaderKey  = "x-goog-api-client"
+	downloadMany    = "download_many"
+	downloadSharded = "download_sharded"
+)
+
+// Sets invocation ID headers on the context which will be propagated as
+// headers in the call to the service (for both gRPC and HTTP).
+func setUsageMetricHeader(ctx context.Context, method string) context.Context {
+	header := fmt.Sprintf("gccl-gcs-cmd/tm.%s", method)
```
Comment (tritone) [inline storage/transfermanager/downloader.go:323]: nit: I'd also make the key a const.

File: storage/invoke.go [hunk 2]
```diff
@@ -84,7 +84,21 @@ func setInvocationHeaders(ctx context.Context, invocationID string, attempts int
...
+		xGoogHeader = strings.Join([]string{xGoogHeader, v}, " ")
+	}
+
+	if hdrs[xGoogHeaderKey] != nil {
+		// Replace the key instead of adding it, if there was anything to merge with.
+		hdrs[xGoogHeaderKey] = []string{xGoogHeader}
+	} else {
+		// TODO: keep this line when removing the above code.
+		ctx = callctx.SetHeaders(ctx, xGoogHeaderKey, xGoogHeader)
+	}
```
Comment (tritone) [inline storage/invoke.go:100]: This looks right but could you add a test? Possibly could be added as a case in the existing tests for invoke, but if a separate one is easier that's fine too.

**Review by tritone — COMMENTED** [review]
Just needs a unit test, otherwise looks good.

File: storage/invoke.go (hunk 2 above)
Comment (BrennaEpp) [reply storage/invoke.go:100]: Added. I think a separate one is better because it should be removed with the code.

File: storage/invoke_test.go [hunk 3]
```diff
@@ -298,6 +298,68 @@ func (f *fakeApiaryRequest) Header() http.Header {
 	return f.header
 }
 
+func TestInvokeHeaderMerge(t *testing.T) {
```
Comment (tritone) [inline storage/invoke_test.go:301]: Could you add a comment explaining what this is testing for? Probably a little unintuitive to anyone who is not familiar.

**Review by tritone — APPROVED** [review]
LGTM, just one more suggestion.

**Review by tritone — APPROVED** [review]

**Pull request merged**